src/world.rs

```diff
@@ -0,0 +1,128 @@
+use crate::{
+    rollbacks::{Rollbackable, RollbackableHashMap, RollbackableVec},
+    store::{ContractStorage, InitialStorage, StorageError, StorageKey},
+};
+use std::{cell::RefCell, collections::HashMap, rc::Rc};
+use u256::{H160, U256};
+
+#[derive(Debug, PartialEq, Clone, Default)]
+pub struct L2ToL1Log {
+    pub key: U256,
+    pub value: U256,
+    pub is_service: bool,
+    pub address: H160,
+    pub shard_id: u8,
+    pub tx_number: u16,
+}
+
+#[derive(Debug, PartialEq, Default, Clone)]
+pub struct Event {
+    pub key: U256,
+    pub value: U256,
+    pub is_first: bool,
+    pub shard_id: u8,
+    pub tx_number: u16,
+}
+
+#[derive(Debug)]
+pub struct World {
```

Comment:
I wouldn't call this `World` as it might lead to confusion since this struct includes not only aspects related to the general state of the chain but also some internal VM state. Could we change the name to smth more related to storage? Perhaps `StateStorage` or `VMStorage` would be more appropriate. Additionally, we should update the name of `WorldSnapshot` to be consistent with this change. Feel free to propose other names if you have a different idea—these were just the first that came to mind.